### DISCUSSION TITLE
Default spacings

### DISCUSSION THREAD

**Discussion by PubliAlex** [OFFLINE-GmbH/oc-boxes-support#163] [Help]
Hello,

By reading the doc, It seems it's not possible actually to define default values for spacing.

This is a feature I would need, add default "medium" spacing to my boxes

Thank you

**Comment by tobias-kuendig** [comment]
You can define your own spacings:
https://docs.boxes.offline.ch/use-cases/spacing.html#defining-the-scale

You could add your custom "default" class to the "none" spacing. The naming does not make much sense in this case, but it should get you the default spacing class for all boxes that currently have "none" set.

Another option is to add a custom event listener for when a box is saved and force your spacing there:

```php
// Provider.php
use OFFLINE\Boxes\Models\Box;

public function boot()
{
    // Untested code!
    Box::extend(function (Box $box) {
        $box->bindEvent('model.beforeSave', function () use ($box) {
            if (!$box->data[Box::SPACING_KEY_BEFORE]) {
                $box->data[Box::SPACING_KEY_BEFORE] = 'your-default-spacing';
            }
        });
    });
}
```
